src/rdf4cpp/rdf/datatypes/registry/util/CharConvExt.hpp

```diff
@@ -0,0 +1,166 @@
+    }
+}
+
+/**
+ * Serializes an integral type into its (SPARQL) canonical representation
+ * see https://www.w3.org/TR/2004/REC-xmlschema-2-20041028/#dt-integer
+ *
+ * @param value the value to be serialized
+ */
+template<std::integral I>
+std::string to_chars(I const value) noexcept {
+    // +1 because of definition of digits10 https://en.cppreference.com/w/cpp/types/numeric_limits/digits10
+    // +1 for sign
+    static constexpr size_t buf_sz = std::numeric_limits<I>::digits10 + 1 + static_cast<size_t>(std::is_signed_v<I>);
+
+    std::array<char, buf_sz> buf;
+    std::to_chars_result const res = std::to_chars(buf.data(), buf.data() + buf.size(), value);
+
+    assert(res.ec == std::errc{});
+
+    return std::string{buf.data(), res.ptr};
+}
+
+/**
+ * Parses a valid string representation of a floating point number
+ *
+ * @tparam F the result floating point type
+ * @param s the string to be parsed
+ * @return the resulting value
+ * @throws std::runtime_error if the string cannot be parsed
+ */
+template<std::floating_point F>
+F from_chars(std::string_view s) {
+    if (s.starts_with('+')) {
+        // from_chars does not allow initial +
+        s.remove_prefix(1);
+    }
+
+    F value;
+    std::from_chars_result const res = std::from_chars(s.data(), s.data() + s.size(), value, std::chars_format::general);
+
+    if (res.ec != std::errc{} || res.ptr != s.data() + s.size()) {
+        // parsing did not reach end of string => it contains invalid characters
+        throw std::runtime_error{"xsd floating point parsing error: " + std::make_error_code(res.ec).message()};
+    }
+
+    return value;
+}
+
+namespace detail  {
+/**
+ * equivalent to static_cast<size_t>(1 + std::log10(value))
+ * only exists because the above is not a constexpr in clang
+ */
+template<typename T>
+constexpr size_t log10ceil(T const value) noexcept {
+    if (value < 10) {
+        return 1;
+    }
+    return 1 + log10ceil(value / 10);
+}
+} // namespace detail
+
+/**
+ * Serializes a floating point number into its (SPARQL) canonical string representation
+ * see https://www.w3.org/TR/2004/REC-xmlschema-2-20041028/#dt-float
+ *
+ * @param value the value to be serialized
+ */
+template<std::floating_point F>
+std::string to_chars(F const value) noexcept {
+    if (std::isnan(value)) {
+        return "NaN";
+    }
+
+    if (std::isinf(value)) {
+        if (value > 0) {
+            return "INF";
+        } else {
+            return "-INF";
+        }
+    }
+
+    // +1 for minus in mantissa
+    // +1 for integral part
+    // +1 for dot
+    // +1 for E
+    // +1 for minus in exponent
+    // at least 2 for exponent because the (c++) standard says so (https://en.cppreference.com/w/cpp/utility/to_chars)
+    static constexpr size_t buf_sz = 5 + std::numeric_limits<F>::max_digits10 + std::max(2ul, detail::log10ceil(std::numeric_limits<F>::max_exponent10));
+    std::array<char, buf_sz> buf;
+
+    std::to_chars_result res = [&]() {
```

Comment:
is noexcept automatically applied here because the enclosing function is noexcept? otherwise you could also make the lambda noexcept?

Reply:
Sadly, no: https://godbolt.org/z/fz3MbxdMe
Thing is: those are immediatly invoked lambdas, not sure how to test how the compiler treats them

Added it to #84